tsfc/driver.py

```diff
@@ -34,6 +34,38 @@
 sys.setrecursionlimit(3000)
 
 
+class TSFCIntegralDataInfo(object):
+    __slots__ = ("domain", "integral_type", "subdomain_id", "domain_number",
+                 "arguments",
+                 "coefficients", "coefficient_numbers")
+    """A minimal set of objects for kernel builders.
+
+    :arg domain: The mesh.
+    :arg integral_type: The type of integral.
+    :arg subdomain_id: What is the subdomain id for this kernel.
+    :arg domain_number: Which domain number in the original form
+        does this kernel correspond to (can be used to index into
+        original_form.ufl_domains() to get the correct domain).
+    :arg coefficients: A list of coefficients.
+    :arg coefficient_numbers: A list of which coefficients from the
+        form the kernel needs.
+
+    This is a minimal set of objects that kernel builders need to
+    construct a kernel from :attr:`integrals` of :class:`~ufl.IntegralData`.
+    """
+    def __init__(self, domain, integral_type, subdomain_id, domain_number,
+                 arguments,
+                 coefficients, coefficient_numbers):
+        self.domain = domain
+        self.integral_type = integral_type
+        self.subdomain_id = subdomain_id
+        self.arguments = arguments
+        self.domain_number = domain_number
+        self.coefficients = coefficients
+        self.coefficient_numbers = coefficient_numbers
+        super(TSFCIntegralDataInfo, self).__init__()
```

Comment:
If this is an immutable object does it make sense for it just to be a `NamedTuple`?